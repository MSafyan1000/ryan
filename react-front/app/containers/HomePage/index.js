/*
 * HomePage
 *
 * This is the first thing users see of our App, at the '/' route
 *
 */

import PropTypes from 'prop-types';
import React from 'react';
import { useQuery } from '@apollo/react-hooks';
import { gql } from 'apollo-boost';
import { FormattedMessage } from 'react-intl';
import messages from './messages';

export const QUERY_ORDERS = gql`
  {
    allOrders {
      id
      description
      total
    }
  }
`;

export default function HomePage() {
  const props = useQuery(QUERY_ORDERS);
  const { loading, error, data } = props;

  if (loading) return <p>Loading...</p>;
  if (error) return <p>Error :(</p>;

  return (
    <>
      {' '}
      <div className="content content-fixed bd-b">
        <div className="container pd-x-0 pd-lg-x-10 pd-xl-x-0">
          <div className="d-sm-flex align-items-center justify-content-between">
            <div>
              <nav aria-label="breadcrumb">
                <ol className="breadcrumb breadcrumb-style1 mg-b-10">
                  <li className="breadcrumb-item">
                    <a href="#">Profile</a>
                  </li>
                  <li className="breadcrumb-item">
                    <a href="#">Discover</a>
                  </li>
                  <li className="breadcrumb-item active" aria-current="page">
                    Groups
                  </li>
                </ol>
              </nav>
              <h4 className="mg-b-0">Discover Groups</h4>
            </div>
            <div className="search-form mg-t-20 mg-sm-t-0">
              <input
                type="search"
                className="form-control"
                placeholder="Search groups"
              />
              <button className="btn" type="button">
                <i data-feather="search" />
              </button>
            </div>
          </div>
        </div>
      </div>
      <div className="content">
        <div className="container pd-x-0 pd-lg-x-10 pd-xl-x-0">
          <div className="row">
            <div className="col-lg-9">
              <div className="row row-xs mg-b-25">
                {/* col */}
                <div className="col-sm-6 col-md-4 mg-t-10">
                  <ul className="list-group">
                    <li className="list-group-item d-flex justify-content-between align-items-center">
                      Cras justo odio
                      <span className="badge badge-primary badge-pill">14</span>
                    </li>
                    <li className="list-group-item d-flex justify-content-between align-items-center">
                      Dapibus ac facilisis in
                      <span className="badge badge-primary badge-pill">2</span>
                    </li>
                    <li className="list-group-item d-flex justify-content-between align-items-center">
                      Morbi leo risus
                      <span className="badge badge-primary badge-pill">1</span>
                    </li>
                  </ul>
                </div>
                {/* col */}
              </div>
              {/* row */}
            </div>
            {/* col */}
            <div className="col-lg-3 mg-t-40 mg-lg-t-0">
              <h6 className="">Groups By Location</h6>
              <ul className="list-group">
                <li className="list-group-item">Cras justo odio</li>
                <li className="list-group-item">Dapibus ac facilisis in</li>
                <li className="list-group-item">Morbi leo risus</li>
                <li className="list-group-item">Porta ac consectetur ac</li>
                <li className="list-group-item">Vestibulum at eros</li>
              </ul>
            </div>
            {/* col */}
          </div>
          {/* row */}
        </div>
        {/* container */}
      </div>
    </>
  );

  return (
    <div>
      <h1>
        <FormattedMessage {...messages.header} />
      </h1>
      <div data-test="orders">
        {data.allOrders.map(({ description, total, id }) => (
          <div key={id}>
            <p>
              {description}: {total}
            </p>
          </div>
        ))}
      </div>
    </div>
  );
}

HomePage.propTypes = {
  loading: PropTypes.bool,
  error: PropTypes.object,
  data: PropTypes.array,
};
