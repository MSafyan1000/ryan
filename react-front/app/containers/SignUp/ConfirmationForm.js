import React from 'react';

const ConfirmationForm = () => {
  const [email, setEmail] = useState('');
  const [confirmationCode, setConfirmationCode] = useState('');

  function validateForm() {
    return email.length > 0 && confirmationCode.length > 0;
  }

  async function handleSubmit(event) {
    event.preventDefault();

    try {
      await Auth.confirmSignUp(email, confirmationCode);
      // setEmail('');
      props.history.push('/go');
    } catch (e) {
      console.log(e);
      alert(e.message);
    }
  }

  return (
    <form onSubmit={handleSubmit} data-testid="form">
      <div className="pd-t-20 wd-100p">
        <h4 className="tx-color-01 mg-b-5">Confirmation Page</h4>
        <div className="form-group">
          <label>Email address</label>
          <input
            type="email"
            className="form-control"
            placeholder="Enter your email address"
            data-testid="inputEmail"
            value={email}
            onChange={e => setEmail(e.target.value)}
          />
        </div>
        <div className="form-group">
          <div className="d-flex justify-content-between mg-b-5">
            <label className="mg-b-0-f">Confirmation Code</label>
          </div>
          <input
            className="form-control"
            placeholder="Enter your Code send via Email"
            data-testid="inputPassword"
            onChange={e => setConfirmationCode(e.target.value)}
            type="password"
          />
        </div>

        <div className="form-group tx-12">
          By clicking <strong>Create an account</strong> below, you agree to our
          terms of service and privacy statement.
        </div>
        {/* form-group */}
        <button
          className="btn btn-brand-02 btn-block"
          data-testid="submitButton"
          disabled={!validateForm()}
          onClick={handleSubmit}
        >
          Confirm
        </button>
        <div className="divider-text">or</div>
        <button className="btn btn-outline-facebook btn-block">
          Sign Up With Facebook
        </button>
        <button className="btn btn-outline-twitter btn-block">
          Sign Up With Twitter
        </button>
        <div className="tx-13 mg-t-20 tx-center">
          Already have an account? <Link to="/sign_in">Sign In</Link>
        </div>
      </div>
    </form>
  );
};

export default withRouter(ConfirmationForm);
