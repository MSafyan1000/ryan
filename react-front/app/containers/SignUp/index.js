/* eslint-disable no-console */
/* eslint-disable no-alert */
/*
 * HomePage
 *
 * This is the first thing users see of our App, at the '/' route
 *
 */

import Header from '../../components/Header';
import Footer from '../../components/Footer';
import React, { useState, useEffect } from 'react';
import '../../../THEME/main/assets/css/dashforge.auth.css';
import { Link, withRouter } from 'react-router-dom';
import { Auth } from 'aws-amplify';

// import SignUpForm from './signUpForm';

function HomePage() {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [confirmationCode, setConfirmationCode] = useState('');
  const [signedUp, setSignedUp] = useState(false);

  useEffect(() => {}, [signedUp]);

  function validateForm() {
    return email.length > 0 && password.length > 0;
  }

  async function handleSubmit(event) {
    event.preventDefault();
    if (!signedUp) {
      try {
        await Auth.signUp(email, password);
        setSignedUp(true);
        props.history.push('/sign_up');
      } catch (e) {
        console.log(e);
        if (e.message && e.name === 'UserNotConfirmedException') {
          setSignedUp(true);
          console.log('from signedUp error');
        }
        alert(e.message);
      }
    } else {
      try {
        await Auth.confirmSignUp(email, confirmationCode);
        console.log('auth confirmCode');
        props.history.push('/profile');
      } catch (e) {
        console.log(e);
        alert(e.message);
      }
    }
  }
  return (
    <>
      <div>
        <Header />
        {/* navbar */}
        {/* {console.table('code', confirmationCode, 'email', email, password)} */}
        <div className="content content-fixed content-auth">
          <div className="container">
            <div className="media align-items-stretch justify-content-center ht-100p">
              <div className="sign-wrapper mg-lg-r-50 mg-xl-r-60">
                {!signedUp ? (
                  <form onSubmit={handleSubmit} data-testid="form">
                    <div className="pd-t-20 wd-100p">
                      <h4 className="tx-color-01 mg-b-5">Create New Account</h4>
                      <p className="tx-color-03 tx-16 mg-b-40">
                        Its free to signup and only takes a minute.
                      </p>
                      <div className="form-group">
                        <label>Email address</label>
                        <input
                          type="email"
                          className="form-control"
                          placeholder="Enter your email address"
                          data-testid="inputEmail"
                          value={email}
                          onChange={e => setEmail(e.target.value)}
                        />
                      </div>
                      <div className="form-group">
                        <div className="d-flex justify-content-between mg-b-5">
                          <label className="mg-b-0-f">Password</label>
                        </div>
                        <input
                          className="form-control"
                          placeholder="Enter your password"
                          data-testid="inputPassword"
                          onChange={e => setPassword(e.target.value)}
                          type="password"
                        />
                      </div>
                      <div className="form-group">
                        <label>Firstname</label>
                        <input
                          type="text"
                          className="form-control"
                          data-testid="inputFirstName"
                          placeholder="Enter your firstname"
                        />
                      </div>
                      <div className="form-group">
                        <label>Lastname</label>
                        <input
                          type="text"
                          className="form-control"
                          data-testid="inputLastName"
                          placeholder="Enter your lastname"
                        />
                      </div>
                      <div className="form-group tx-12">
                        By clicking <strong>Create an account</strong> below,
                        you agree to our terms of service and privacy statement.
                      </div>
                      {/* form-group */}
                      <button
                        className="btn btn-brand-02 btn-block"
                        data-testid="submitButton"
                        disabled={!validateForm()}
                        onClick={handleSubmit}
                      >
                        Create Account
                      </button>
                      <div className="divider-text">or</div>
                      <button className="btn btn-outline-facebook btn-block">
                        Sign Up With Facebook
                      </button>
                      <button className="btn btn-outline-twitter btn-block">
                        Sign Up With Twitter
                      </button>
                      <div className="tx-13 mg-t-20 tx-center">
                        Already have an account?{' '}
                        <Link to="/sign_in">Sign In</Link>
                      </div>
                    </div>
                  </form>
                ) : (
                  <>
                    {' '}
                    <form onSubmit={handleSubmit} data-testid="form">
                      <div className="pd-t-20 wd-100p">
                        <h4 className="tx-color-01 mg-b-5">
                          Confirmation Page
                        </h4>
                        <div className="form-group">
                          <label>Email address</label>
                          <input
                            type="email"
                            className="form-control"
                            placeholder="Enter your email address"
                            data-testid="inputEmail"
                            value={email}
                            onChange={e => setEmail(e.target.value)}
                          />
                        </div>
                        <div className="form-group">
                          <div className="d-flex justify-content-between mg-b-5">
                            <label className="mg-b-0-f">
                              Confirmation Code
                            </label>
                          </div>
                          <input
                            className="form-control"
                            placeholder="Enter your Code send via Email"
                            data-testid="inputPassword"
                            onChange={e => setConfirmationCode(e.target.value)}
                            type="password"
                          />
                        </div>

                        <div className="form-group tx-12">
                          By clicking <strong>Create an account</strong> below,
                          you agree to our terms of service and privacy
                          statement.
                        </div>
                        {/* form-group */}
                        <button
                          className="btn btn-brand-02 btn-block"
                          data-testid="submitButton"
                          onClick={handleSubmit}
                        >
                          Confirm
                        </button>
                        <div className="divider-text">or</div>
                        <button className="btn btn-outline-facebook btn-block">
                          Sign Up With Facebook
                        </button>
                        <button className="btn btn-outline-twitter btn-block">
                          Sign Up With Twitter
                        </button>
                        <div className="tx-13 mg-t-20 tx-center">
                          Already have an account?{' '}
                          <Link to="/sign_in">Sign In</Link>
                        </div>
                      </div>
                    </form>
                  </>
                )}
              </div>
              {/* sign-wrapper */}
              <div className="media-body pd-y-30 pd-lg-x-50 pd-xl-x-60 align-items-center d-none d-lg-flex pos-relative">
                <div className="mx-lg-wd-500 mx-xl-wd-550">
                  <img
                    src="../../assets/img/img16.png"
                    className="img-fluid"
                    alt=""
                  />
                </div>
                <div className="pos-absolute b-0 r-0 tx-12">
                  Social media marketing vector is created by{' '}
                  <a
                    href="https://www.freepik.com/pikisuperstar"
                    target="_blank"
                  >
                    pikisuperstar (freepik.com)
                  </a>
                </div>
              </div>
              {/* media-body */}
            </div>
            {/* media */}
          </div>
          {/* container */}
        </div>
        {/* content */}
        <Footer />
      </div>
    </>
  );
}

export default withRouter(HomePage);
// export const HomePageTest = HomePage;
