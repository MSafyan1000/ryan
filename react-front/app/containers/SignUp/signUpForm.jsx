import React, { useState } from 'react';
import { Auth } from 'aws-amplify';
import { Link, withRouter } from 'react-router-dom';

const SignUpForm = () => {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');

  function validateForm() {
    return email.length > 0 && password.length > 0;
  }

  async function handleSubmit(event) {
    event.preventDefault();

    try {
      await Auth.signUp(email, password);
      setEmail('');
      // props.history.push('/sign_in');
    } catch (e) {
      console.log(e);
      alert(e.message);
    }
  }

  return (
    <form onSubmit={handleSubmit} data-testid="form">
      <div className="pd-t-20 wd-100p">
        <h4 className="tx-color-01 mg-b-5">Create New Account</h4>
        <p className="tx-color-03 tx-16 mg-b-40">
          Its free to signup and only takes a minute.
        </p>
        <div className="form-group">
          <label>Email address</label>
          <input
            type="email"
            className="form-control"
            placeholder="Enter your email address"
            data-testid="inputEmail"
            value={email}
            onChange={e => setEmail(e.target.value)}
          />
        </div>
        <div className="form-group">
          <div className="d-flex justify-content-between mg-b-5">
            <label className="mg-b-0-f">Password</label>
          </div>
          <input
            className="form-control"
            placeholder="Enter your password"
            data-testid="inputPassword"
            onChange={e => setPassword(e.target.value)}
            type="password"
          />
        </div>
        <div className="form-group">
          <label>Firstname</label>
          <input
            type="text"
            className="form-control"
            data-testid="inputFirstName"
            placeholder="Enter your firstname"
          />
        </div>
        <div className="form-group">
          <label>Lastname</label>
          <input
            type="text"
            className="form-control"
            data-testid="inputLastName"
            placeholder="Enter your lastname"
          />
        </div>
        <div className="form-group tx-12">
          By clicking <strong>Create an account</strong> below, you agree to our
          terms of service and privacy statement.
        </div>
        {/* form-group */}
        <button
          className="btn btn-brand-02 btn-block"
          data-testid="submitButton"
          disabled={!validateForm()}
          onClick={handleSubmit}
        >
          Create Account
        </button>
        <div className="divider-text">or</div>
        <button className="btn btn-outline-facebook btn-block">
          Sign Up With Facebook
        </button>
        <button className="btn btn-outline-twitter btn-block">
          Sign Up With Twitter
        </button>
        <div className="tx-13 mg-t-20 tx-center">
          Already have an account? <Link to="/sign_in">Sign In</Link>
        </div>
      </div>
    </form>
  );
};

export const handleSubmit = handleSubmit;
export default withRouter(SignUpForm);
