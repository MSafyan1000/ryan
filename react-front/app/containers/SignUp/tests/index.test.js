// import { MockedProvider } from '@apollo/react-testing';
// import { IntlProvider } from 'react-intl';

import Enzyme, { shallow, mount } from 'enzyme';
import React from 'react';
import HomePage, { HomePageTest } from '../index';

import {
  cleanup,
  render,
  fireEvent,
  waitForDomChange,
} from '@testing-library/react';
import Adapter from 'enzyme-adapter-react-16';
import { createSerializer } from 'enzyme-to-json';
import { Auth, input } from 'aws-amplify';
import { MemoryRouter } from 'react-router-dom';
import SignUpPage from '../index';
import SignUp, { handleSubmit } from '../signUpForm';

expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

Enzyme.configure({ adapter: new Adapter() });

describe('react testing vs Enzyme', () => {
  afterEach(cleanup);

  describe('Enzyme', () => {
    //these are for enzyme
    const wrapper = mount(
      <MemoryRouter>
        <SignUp />
      </MemoryRouter>,
    );
    const inputEmail = wrapper.find('[data-testid="inputEmail"]');
    const inputPassword = wrapper.find('[data-testid="inputPassword"]');
    const inputFirstName = wrapper.find('[data-testid="inputFirstName"]');
    const inputLastName = wrapper.find('[data-testid="inputLastName"]');
    const submitButton = wrapper.find('[data-testid="submitButton"]');

    const mockHandleSubmit = jest.fn();
    Auth.signUp = jest.fn();
    global.alert = jest.fn();

    //these are for react-testing-library
    const { getByTestId, debug } = render(
      <MemoryRouter>
        <SignUp />
      </MemoryRouter>,
    );
    const inputEmailT = getByTestId('inputEmail');
    const inputPasswordT = getByTestId('inputPassword');
    const inputFirstNameT = getByTestId('inputFirstName');
    const inputLastNameT = getByTestId('inputLastName');
    const submitButtonT = getByTestId('submitButton');

    it('snapshot SignUp', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('render form fields', async () => {
      expect(inputEmail.length).toBe(1);
      expect(inputPassword.length).toBe(1);
      expect(inputFirstName.length).toBe(1);
      expect(inputLastName.length).toBe(1);
      expect(submitButton.length).toBe(1);
    });

    it('check weather form fields are empty', () => {
      expect(inputEmail.text()).toBe('');
      expect(inputPassword.text()).toBe('');
      expect(inputFirstName.text()).toBe('');
      expect(inputLastName.text()).toBe('');
      expect(submitButton.prop('disabled')).toBeTruthy();
    });

    it('check Auth.SingUp in triggered', () => {
      submitButton.simulate('submit');
      expect(Auth.signUp).toBeCalled();
    });

    it('check Alert reject Auth.signUp', () => {
      window.alert = jest.fn();
      Auth.signUp = jest.fn(() => {
        throw new Error();
      });
      submitButton.simulate('submit');
      expect(Auth.signUp).toBeCalled();
      expect(window.alert).toBeCalled();
    });

    it('rendering footer and headers', () => {
      const wrapper = mount(
        <MemoryRouter>
          <SignUpPage />
        </MemoryRouter>,
      );

      const Header = wrapper.find('Header');
      const Footer = wrapper.find('Footer');
      expect(Header.length).toBe(1);
      expect(Footer.length).toBe(1);
    });

    it('check form fields onChange', () => {
      wrapper.setProps;

      fireEvent.change(inputEmailT, { target: { value: 'Email' } });
      fireEvent.change(inputPasswordT, { target: { value: 'Password' } });
      fireEvent.change(inputFirstNameT, { target: { value: 'FirstName' } });
      fireEvent.change(inputLastNameT, { target: { value: 'LastName' } });

      expect(inputEmailT.value).toBe('Email');
      expect(inputPasswordT.value).toBe('Password');
      expect(inputFirstNameT.value).toBe('FirstName');
      expect(inputLastNameT.value).toBe('LastName');
    });

    // it('check button password validation', () => {
    //   wrapper.setState({ email: 'email', password: 'password' });
    //   submitButton.simulate('submit');
    //   wrapper.update();
    //   wrapper.setProps({});
    //   // expect(wrapper.update);
    //   expect(submitButton.props()).toBe({});
    // });

    // it('empty form field after submit', () => {
    //   fireEvent.change(inputEmailT, { target: { value: 'Email' } });
    //   submitButton.simulate('submit');
    //   expect(Auth.signUp).toBeCalled();
    //   expect(inputEmailT.value).toBe('');
    // });
  });

  describe('testingLib', () => {
    it('testingLib', async () => {
      const handleSubmit = jest.fn();

      const { getByTestId, debug } = render(
        <MemoryRouter>
          <SignUp submit={handleSubmit} />
        </MemoryRouter>,
      );

      const inputEmail = getByTestId('inputEmail');
      const submitButton = getByTestId('submitButton');
      const form = getByTestId('form');

      Auth.signUp = jest.fn();

      fireEvent.change(inputEmail, { target: { value: 'Email' } });

      expect(inputEmail.value).toBe('Email');
      // submitButton.simulate('click', { handler });

      // await waitFor();
      // fireEvent.click(submitButton, { handleSubmit });
      fireEvent.submit(submitButton);
      fireEvent.submit(form);

      // expect(submitButton).toBe('');
      // debug();

      expect(Auth.signUp).toHaveBeenCalledTimes(2);
      // expect(handleSubmit).toHaveBeenCalledTimes(1);

      expect(inputEmail.value).toBe('Email');
    });
  });
});

// it('calls event handler; "handleSubmit"', async () => {
//   const Wrapper = shallow(<SignUp />);
//   // const inputEmail = findByTestAttr(Wrapper, 'inputEmail');
//   const inputEmail = Wrapper.find('.inputEmailTest').simulate('change');
//   expect(inputEmail).toHaveBeenCalled();

// const componentInstance = Wrapper.instance();

// componentInstance.setState({
//   isLoading: false,
//   email: 'demo',
//   password: 'demo',
// });
// const event = {
//   preventDefault: () => {},
// };
// Auth.signIn = jest.fn().mockImplementation(() => {
//   message: 'success';
// });
// await componentInstance.handleSubmit(event);
// expect(componentInstance.state.isLoading).toEqual(true);
// });

// describe('Ben Awad', () => {
//   afterEach(cleanup);
//   it('react testing', async () => {
//     const push = jest.fn();
//     const { getByTestId, debug, waitForDomChange, waitForElement } = render(
//       <MemoryRouter>
//         <HomePage />
//       </MemoryRouter>,
//     );
//     const inputEmail = getByTestId('inputEmail');
//     const submitButton = getByTestId('submitButton');
//     fireEvent.change(inputEmail, { target: { value: 'email' } });
//     fireEvent.submit(submitButton);
//     await waitForElement(() => getByText('email'));
//     await waitForDomChange();
//     debug();
//   });
// });

// it('mocking signUp via andre', () => {
//   const mockSignUp = jest.fn().mockReturnValue(
//     Promise.resolve({
//       message: 'success',
//     }),
//   );
//   expect(render(<SignUp />).find('[data-test=inputEmail'));
// });

// describe('<SignUp />', () => {
//   it('should render table of currencies', async () => {
//     const wrapper = mount(
//       <MockedProvider mocks={[]} addTypename={false}>
//         <IntlProvider locale="en">
//           <SignUp />
//         </IntlProvider>
//       </MockedProvider>,
//     );
//     expect(wrapper.find('header')).toHaveLength(1);
//   });
// });

// describe('testing SignUp', () => {
//   it('check submitHandle is trigger when button is clicked', () => {
//     const spy = sinon.spy();
//     const wrapper = shallow(<SignUp />);
//     wrapper.find('.btn-brand-02').simulate('click');
//     expect(spy.calledOnce).toBe(true);
//   });
// });

// it('mocking api request', () => {
//   global.Auth.SignUp = jest.fn(() =>
//     Promise.resolve({
//       json: () => Promise.resolve({ data: { message: 'success' } }),
//     }),
//   );
// });

// it('andre testing state', () => {
//   // expect.assertions(1);
//   expect(shallow(<SignUp />)).toMatchSnapshot();
// });

// it("rendering props Component",()=>{
//   const wrapper=shallow(<SignUp prop1="name" prop2={object={key1:"hello"}} toggleForcast={()=>{}} />)

//   expect(wrapper).toMatchSnapshot()
// })

// it("rendering props Component",()=>{
//   const wrapper=render(<SignUp prop1="name" prop2={object={key1:"hello"}} toggleForcast={()=>{}} />)

//   expect(wrapper).toMatchSnapshot()
// })

// it('get a div inside component',()=>{
//   const wrapper=render(<SignUp prop1="name" prop2={object={key1:"hello"}} toggleForcast={()=>{}} />)
// //you can use any class or id for selecting component inside wrapper
//   const text=wrapper.find('LargeText')

//   const wrapper=mount(<SignUp prop1="name" prop2={object={key1:"hello"}} toggleForcast={spy} />)

// })

// it('calls a function on click',()=>{
//   const spy=sinon.spy();

//   const wrapper=mount(<SignUp prop1="name" prop2={object={key1:"hello"}} toggleForcast={spy} />)

//   wrapper.find("div").first().simulate("click");

//   expect.(spy.calledOnce).toBe(true);

// })

// it('2+2', () => {
//   expect(2 + 2).toBe(4);
// });

// it('mock signIN', () => {
//   const mockSignIn = jest.fn().mockReturnValue(
//     Promise.resolve({
//       json: () =>
//         Promise.resolve({
//           username: 'userName',
//           password: 'userPassword',
//         }),
//     }),
//   );
//   return mockSignIn;
// });

global.React = React;
global.shallow = shallow;
global.render = render;
global.mount = mount;
global.sinon = sinon;
