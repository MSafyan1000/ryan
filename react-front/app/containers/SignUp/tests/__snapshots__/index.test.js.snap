// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react testing vs Enzyme Enzyme snapshot SignUp 1`] = `
<form
  data-testid="form"
  onSubmit={[Function]}
>
  <div
    className="pd-t-20 wd-100p"
  >
    <h4
      className="tx-color-01 mg-b-5"
    >
      Create New Account
    </h4>
    <p
      className="tx-color-03 tx-16 mg-b-40"
    >
      Its free to signup and only takes a minute.
    </p>
    <div
      className="form-group"
    >
      <label>
        Email address
      </label>
      <input
        className="form-control"
        data-testid="inputEmail"
        onChange={[Function]}
        placeholder="Enter your email address"
        type="email"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <div
        className="d-flex justify-content-between mg-b-5"
      >
        <label
          className="mg-b-0-f"
        >
          Password
        </label>
      </div>
      <input
        className="form-control"
        data-testid="inputPassword"
        placeholder="Enter your password"
        type="password"
      />
    </div>
    <div
      className="form-group"
    >
      <label>
        Firstname
      </label>
      <input
        className="form-control"
        data-testid="inputFirstName"
        placeholder="Enter your firstname"
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <label>
        Lastname
      </label>
      <input
        className="form-control"
        data-testid="inputLastName"
        placeholder="Enter your lastname"
        type="text"
      />
    </div>
    <div
      className="form-group tx-12"
    >
      By clicking 
      <strong>
        Create an account
      </strong>
       below, you agree to our terms of service and privacy statement.
    </div>
    <button
      className="btn btn-brand-02 btn-block"
      data-testid="submitButton"
      disabled={true}
      onClick={[Function]}
    >
      Create Account
    </button>
    <div
      className="divider-text"
    >
      or
    </div>
    <button
      className="btn btn-outline-facebook btn-block"
    >
      Sign Up With Facebook
    </button>
    <button
      className="btn btn-outline-twitter btn-block"
    >
      Sign Up With Twitter
    </button>
    <div
      className="tx-13 mg-t-20 tx-center"
    >
      Already have an account? 
      <a
        href="/sign_in"
        onClick={[Function]}
      >
        Sign In
      </a>
    </div>
  </div>
</form>
`;
